### PR TITLE
Lesser Angels

### PR DESCRIPTION
**Pull request by r4d6**
**Warning! This is a _work in progress_ PR, made by a first time coder. I don't expect it to be added anytime soon, but I would like some help**

## About The Pull Request

This add "Lesser Angels" as a round start race
Lesser Angels are like normal angels, but they don't have the "holy" trait, their wings can burn off and they need more atmosphere to be able to fly. And only in no gravity (Though that still need to be added)

## Why It's Good For The Game

I asked why angels were not in the race selection menu, and was told to "code it myself". As such, I coded it, then was told that since angels were straight up "upgraded humans", that the PR would have been rejected without second thought, thus I decided to make a "lesser angel" race, weaker and with probably a better chance to be accepted.

## Changelog
:cl:
Added Lesser Angels as a starting race
Added Lesser Angels as a race
/:cl:

### CODE REVIEW

**Comment by lolman360** [comment]
>race PR without contacting headcoder

Madman

**Comment by r4d6** [comment]
> > race PR without contacting headcoder
> 
> Madman

Well, I am a new coder, which did this when told "Code it yourself", and I did not know that I needed to contact the headcoder.
Also, This is far from complete, and I would like other coder to give suggestions and point out errors that I did.
This is also basically a patchwork of the Angel race with 1-2 elements from the moth race, mainly the burnt wing code.

**Comment by Trilbyspaceclone** [comment]
@lolman360 on cit you can add new races without contacting anyone
Thats a tg thing

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm [hunk 1]
```diff
@@ -0,0 +1,160 @@
+    
+/datum/species/langel
+	name = "Lesser Angel"
+	id = "langel"
+	default_color = "FFFFFF"
+	species_traits = list(EYECOLOR,HAIR,FACEHAIR,LIPS)
+	mutant_bodyparts = list("wings")
+	default_features = list("mcolor" = "FFF", "tail_human" = "None", "ears" = "None", "wings" = "Angel")
+	use_skintones = 1
+	no_equip = list(SLOT_BACK)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/species_types/lesserangel.dm:10]: This means they will NEVER be able to ware ANYTHING on their back - if that is the case, I love it. If thats not intended, pointing it out

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	blacklisted = 1
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/species_types/lesserangel.dm]: Blacklisted means they can be gotten via magic/mirrors. Giving the drawbacks I think its safe to not blacklist them

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	return 1
+
+
+/datum/species/langel/spec_stun(mob/living/carbon/human/H,amount)
+	if(H.movement_type & FLYING)
+		ToggleFlight(H,0)
+		flyslip(H)
+	. = ..()
+
+## /datum/species/langel/negates_gravity(mob/living/carbon/human/H)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/species_types/lesserangel.dm]: Not to sure why you have ## here

**Review by Trilbyspaceclone — APPROVED** [review]
Love the idea of a race without a backpack

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm (hunk 3 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/human/species_types/lesserangel.dm]: I think that is the part where their wings negate gravity, and for some reason I thought that ## was for comments, don't know why I didn't remember it was //

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm (hunk 1 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/human/species_types/lesserangel.dm:10]: That was already in the base Angel code, and it is confirmed that they can wear satchel

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm (hunk 2 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/human/species_types/lesserangel.dm]: Thanks for informing me of that, if you got more suggestions and code modification idea, you can contact me directly via email at David.Moussette@gmail.com

File: code/modules/mob/living/carbon/human/species_types/lesserangel.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/modules/mob/living/carbon/human/species_types/lesserangel.dm:10]: Odd, noted, but odd

**Comment by deathride58** [comment]
Wouldnt it be better to just add wings as an option for all races? All of the necessary preferences are already in place, it'd just be a single addition to all non-moth roundstart races and a couple of tweaks to the preference menu's code.

Also, flight is a fairly strong ability, there's _very_ good reason it's locked to a race that's only available via xenobio and lavaland loot. Adding it to a roundstart race in its current state will quickly result in the race becoming the meta when there's no sufficient downsides. Standard for roundstart race balance here is that all roundstart races should be either worse or on par with baseline humans (see: mushrooms, plasmamen, and slimes), having a race that's a strict upgrade over humans is a hard no.

This PR's current state has the proposed angel race in a state that's overall mechanically superior to humans. The only downside is that they can't wear things on their back, but that's not that big of an issue when they've still got two hands, a belt slot, and the ability to pull objects. A good way to balance their flight mechanic would be to make flying have its cost be pulled directly from the stamina buffer for all movement, with the sprint buffer going untouched even while sprinting during flight, and sprint during flight resulting in much heavier stamloss drains. But then you'd still have to worry about adding something to make them mechanically interesting to play and more than just "humans but with stam-draining flight", which I don't have any immediate ideas for, but I'm sure you might be able to think of a few if you start throwing things at the wall and seeing what sticks.

@Trilbyspaceclone contributing.md is a thing that we enforce where reasonable. Though in this instance, I think it's safe to say that the PR's author is misguided, confused, and uninformed, so I'll spare them the earful.

**Comment by lolman360** [comment]
just give them moth flash vuln and they'll be literally the same balancewise as moths

**Comment by r4d6** [comment]
@deathride58 
> Wouldnt it be better to just add wings as an option for all races? All of the necessary preferences are already in place, it'd just be a single addition to all non-moth roundstart races and a couple of tweaks to the preference menu's code.
> 
> Also, flight is a fairly strong ability, there's _very_ good reason it's locked to a race that's only available via xenobio and lavaland loot. Adding it to a roundstart race in its current state will quickly result in the race becoming the meta when there's no sufficient downsides. Standard for roundstart race balance here is that all roundstart races should be either worse or on par with baseline humans (see: mushrooms, plasmamen, and slimes), having a race that's a strict upgrade over humans is a hard no.
> 
> This PR's current state has the proposed angel race in a state that's overall mechanically superior to humans. The only downside is that they can't wear things on their back, but that's not that big of an issue when they've still got two hands, a belt slot, and the ability to pull objects. A good way to balance their flight mechanic would be to make flying have its cost be pulled directly from the stamina buffer for all movement, with the sprint buffer going untouched even while sprinting during flight, and sprint during flight resulting in much heavier stamloss drains. But then you'd still have to worry about adding something to make them mechanically interesting to play and more than just "humans but with stam-draining flight", which I don't have any immediate ideas for, but I'm sure you might be able to think of a few if you start throwing things at the wall and seeing what sticks.
> 
> @Trilbyspaceclone contributing.md is a thing that we enforce where reasonable. Though in this instance, I think it's safe to say that the PR's author is misguided, confused, and uninformed, so I'll spare them the earful.

1. I am a first time coder, and I don't have the knowledge or experience to code that.

2. I was already told that before even making this PR, _ It is the entire reason why I made the "Lesser Angel" race instead of just adding one line of code _. Currently, Lesser angels need a 50% atmosphere to fly (Compared to 85% moth and 30% angels), their wings can burn (But I am not sure if the extra condition I put at the line 147 "if(flight && CanFly(H) **&& H.dna.features["wings"] != "Burnt Off"**)" will actually work), I am also planning to prevent their wings from working in gravity, with maybe a message like "You extend your wings and try to fly, but they are not powerful enough to counteract the current gravity". Also I _would_ add it the stamina cost, as it make sense in my opinion, _if_ I had any idea how to do that.

3. As I said, I am a first time coder, and this is practically a patchwork of moth and angel code, with a little bit of value modification.

**Comment by silicons** [comment]
wings for all races god yes
Paging @Poojawa

**Comment by r4d6** [comment]
> wings for all races god yes

Are you able to add non-functional wings as "Tails" in the customize menu @kevinz000 ?
If so I might just close this pr and wait for you to do that. Probably would be simpler too, though I would not know where to start to do this.

**Comment by Poojawa** [comment]
There used to be nonfunctional wings. We still have the sprites also. 

I'm fine with adding more cosmetics to each race. yaya.

**Pull request closed without merging**